modules/configuration-annotation-processor/src/integrationTest/java/org/apache/ignite/internal/configuration/processor/ItProcessorTest.java

```diff
@@ -278,6 +286,122 @@ void testStaticConstants() {
+    }
+
+    /**
+     * Checks that compilation will fail due to misuse of {@link InjectedName}.
+     */
+    @Test
+    void testErrorInjectedNameFieldCodeGeneration() {
+        String packageName = "org.apache.ignite.internal.configuration.processor.injectedname";
+
+        assertThrowsEx(
+                IllegalStateException.class,
+                () -> batchCompile(packageName, "ErrorInjectedName0ConfigurationSchema"),
+                InjectedName.class.getSimpleName()
+        );
+
+        assertThrowsEx(
+                IllegalStateException.class,
+                () -> batchCompile(packageName, "ErrorInjectedName1ConfigurationSchema"),
+                InjectedName.class.getSimpleName()
+        );
+
+        assertThrowsEx(
+                IllegalStateException.class,
+                () -> batchCompile(packageName, "ErrorInjectedName2ConfigurationSchema"),
+                InjectedName.class.getSimpleName()
+        );
+
+        assertThrowsEx(
+                IllegalStateException.class,
+                () -> batchCompile(packageName, "ErrorInjectedName3ConfigurationSchema"),
+                InjectedName.class.getSimpleName()
+        );
+
+        assertThrowsEx(
+                IllegalStateException.class,
+                () -> batchCompile(packageName, "ErrorInjectedName4ConfigurationSchema"),
+                InjectedName.class.getSimpleName()
+        );
+
+        assertThrowsEx(
+                IllegalStateException.class,
+                () -> batchCompile(packageName, "ErrorInjectedName5ConfigurationSchema"),
+                InjectedName.class.getSimpleName()
+        );
+    }
+
+    /**
+     * Checks that compilation will fail due to misuse of {@link Name}.
+     */
+    @Test
+    void testErrorNameFieldCodeGeneration() {
+        String packageName = "org.apache.ignite.internal.configuration.processor.injectedname";
+
+        assertThrowsEx(
+                IllegalStateException.class,
+                () -> batchCompile(packageName, "ErrorName0ConfigurationSchema"),
+                Name.class.getSimpleName()
+        );
+
+        assertThrowsEx(
+                IllegalStateException.class,
+                () -> batchCompile(packageName, "ErrorName1ConfigurationSchema"),
+                Name.class.getSimpleName()
+        );
+    }
+
+    /**
+     * Checks that compilation will succeed when using {@link InjectedName}.
+     */
+    @Test
+    void testSuccessInjectedNameFieldCodeGeneration() {
+        String packageName = "org.apache.ignite.internal.configuration.processor.injectedname";
+
+        ClassName cls0 = ClassName.get(packageName, "SimpleConfigurationSchema");
+        ClassName cls1 = ClassName.get(packageName, "PolyConfigurationSchema");
+
+        BatchCompilation batchCompile = batchCompile(cls0, cls1);
+
+        assertEquals(Compilation.Status.SUCCESS, batchCompile.getCompilationStatus().status());
```

Comment:
It's better to use ` assertThat(batchCompile.getCompilationStatus()).succeededWithoutWarnings()`